### DISCUSSION TITLE
Context Menus

### DISCUSSION THREAD

**Discussion by tcurdt** [fyne-io/fyne#3524] [Q&A] [answered]
I didn't find anything about implementing context menus yet.
Is this supported yet?

Or if not - how does one handle a right-click (to then build a menu at that location)?

Thanks for the help!

**Comment by Jacalz** [comment, marked as answer]
I would suggest having a look at the code handling context menus when right-clicking inside a widget.Entry. https://github.com/fyne-io/fyne/blob/8c2509518b2df442a6b748d9b07754739592e6d7/widget/entry.go#L490

**Reply by Jacalz** [reply]
We don't necessarily have any helpers to make it simpler but you can easily build your own menu like in the example above.

**Reply by tcurdt** [reply]
Let's say I want to add this to a `button` or a `label` - how would you suggest to add that?
Would I extend them by basing off the official impl to extend them with `TappedSecondary`?
Or what's a "fyne way" of doing this?

**Reply by andydotxyz** [reply]
That is bang on. To add functionality to a widget you extend it as you suggest. That way we avoid a huge list of extension points and event handlers on every widget.
